File: src/node2.cpp
#include "Wrapper.h"
#include <iostream>
#include <thread>

using namespace std::literals::chrono_literals;

void c_on_sample(const zenoh::Sample &sample){
    std::cout << "Received " << sample.get_payload().as_string() << std::endl;
}

int main(){

    Wrapper wrapper;

    zenoh::Config conf = zenoh::Config::create_default();
    wrapper.open(conf);

    auto pub = wrapper.declare_publisher("demo/wrapper/2");
    auto sub = wrapper.declare_subscriber("demo/wrapper/1", &c_on_sample, zenoh::closures::none);

    while (1){
        pub.put("Sent from 2");
        std::this_thread::sleep_for(200ms);
    }

    return 0;
}

File: examples/wrapper-nodes/node2.py
import time
import zenoh
from utils.wrapper import Wrapper

def main():

    conf = zenoh.Config()
    # conf.insert_json5("connect/endpoints", '["tcp/127.0.0.1:0"]')

    Wrapper.open(conf)

    pub = Wrapper.declare_publisher("demo/wrapper/2")

    sub = Wrapper.declare_subscriber(
        "demo/wrapper/1",
        lambda sample: print(f"Received: {sample.payload.to_string()}"))
    
    print(f"Node 2 running...")

    # getPeerInfo(session)

    try: 
        while True:
            msg = "Sent from 2"
            pub.put(msg)
            time.sleep(2)
    
    except KeyboardInterrupt:
        pass
    finally:
        sub.undeclare()
        pub.undeclare()

if __name__ == "__main__":
    main()